common/src/main/java/com/wynntils/models/gearinfo/parsing/GearParser.java

```diff
@@ -0,0 +1,211 @@
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Models;
+import com.wynntils.models.concepts.Powder;
+import com.wynntils.models.concepts.Skill;
+import com.wynntils.models.gearinfo.type.GearInfo;
+import com.wynntils.models.gearinfo.type.GearTier;
+import com.wynntils.models.gearinfo.type.GearType;
+import com.wynntils.models.stats.StatCalculator;
+import com.wynntils.models.stats.type.StatActualValue;
+import com.wynntils.models.stats.type.StatPossibleValues;
+import com.wynntils.models.stats.type.StatType;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.type.RangedValue;
+import com.wynntils.utils.wynn.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+
+public final class GearParser {
+    // Test suite: https://regexr.com/776qt
+    public static final Pattern IDENTIFICATION_STAT_PATTERN = Pattern.compile(
+            "^§[ac]([-+]\\d+)(?:§r§[24] to §r§[ac](-?\\d+))?(%| tier|/[35]s)?(?:§r§8/(\\d+)(?:%| tier|/[35]s)?)?(?:§r§2(\\*{1,3}))? ?§r§7 ?(.*)$");
+
+    // Test suite: https://regexr.com/778gh
+    private static final Pattern TIER_AND_REROLL_PATTERN = Pattern.compile(
+            "^(§fNormal|§eUnique|§dRare|§bLegendary|§cFabled|§5Mythic|§aSet|§3Crafted) ([A-Za-z ]+)(?:§r§8)?(?: \\[(\\d+)(?:/(\\d+) Durability)?\\])?$");
+
+    // Test suite: https://regexr.com/778gk
+    private static final Pattern POWDER_PATTERN =
+            Pattern.compile("^§7\\[(\\d+)/(\\d+)\\] Powder Slots(?: \\[§r§(.*)§r§7\\])?$");
+
+    public static GearParseResult parseItemStack(ItemStack itemStack, GearInfo gearInfo) {
+        List<StatActualValue> identifications = new ArrayList<>();
+        List<Powder> powders = new ArrayList<>();
+        int tierCount = 0;
+        int durabilityMax = 0;
+        GearTier tier = null;
+        GearType gearType = null;
+
+        // Parse lore for identifications, powders and rerolls
+        List<Component> lore = ComponentUtils.stripDuplicateBlank(LoreUtils.getTooltipLines(itemStack));
+        lore.remove(0); // remove item name
+
+        for (Component loreLine : lore) {
+            String unformattedLoreLine = WynnUtils.normalizeBadString(loreLine.getString());
+            String coded = ComponentUtils.getCoded(loreLine);
+            String normalizedCoded = WynnUtils.normalizeBadString(coded);
+
+            // Look for powder
+            Matcher powderMatcher = POWDER_PATTERN.matcher(normalizedCoded);
+            if (powderMatcher.matches()) {
+                int usedSlots = Integer.parseInt(powderMatcher.group(1));
+                String codedPowders = powderMatcher.group(3);
+                if (codedPowders == null) continue;
+
+                String powderString = codedPowders.replaceAll("[^✹✦❋❉✤]", "");
```

Comment:
Can't you extract the pattern so it is not "patternized" every time we parse? (I am pretty sure you can do smth like `X_PATTERN.matcher(string).replaceAll(..))`